jalib/jassert.cpp

```diff
@@ -46,6 +43,9 @@ int jassert_quiet = 0;
 static int theLogFileFd = -1;
 static int errConsoleFd = -1;
 
+static jalib::string& tmpDir() {static jalib::string s;return s;};
+static jalib::string& uniquePidStr() {static jalib::string s;return s;};
```

Comment:
I guess the idea here is that the function `tmpDir()` is a method in the jalib class (although to verify this, the reader has to go to the beginning of the file and look for 'namespace').  One could have used the better known paradigm of a field member, `jalib::_tmpdir`, with methods, `jalib::get_tmpdir()` and `jalib::set_tmpdir()`.  Here, we are using an alternative paradigm that is special to C++ only, and allows one to use `tmpDir()` both as a fixed return value (what everybody is familiar with in the standard paradigm), and also as a tricky way to hide a hidden field member (the local variable `static jalib::string s`) and then use the unusual syntex `tmpDir() = ...`.
However, since we will have many developers who are not trained in the intricacies of C++, I propose that we use the better known forms `get_tmpdir()` and `set_tmpdir()`.  Similarly, for `uniquePidStr()` and `theLogFilePath()`.